executor/cpp_executor.go

```diff
@@ -24,78 +27,139 @@ var ErrInvalidVersion = fmt.Errorf("invalid version supplied")
+				Message:    fmt.Sprintf("could not open stdin file in chroot. %s", err.Error()),
 			}
 		}
-		cmd.Stdin = stdin
+		restrictedCmd.Stdin = stdin
 		defer utils.CloseFile(stdin)
 	}
 
-	// Execute command
-	logger.Infof("Executing command [MsgID: %s]", messageID)
-	err = cmd.Run()
-	if err != nil {
-		logger.Errorf("Could not run the command. %s [MsgID: %s]", err.Error(), messageID)
-		return &ExecutionResult{
-			StatusCode: ErInternalError,
-			Message:    fmt.Sprintf("could not run the command. %s", err.Error()),
+	// Execute the command
+	e.logger.Infof("Executing command [MsgID: %s]", messageID)
+	runErr := restrictedCmd.Run()
+	if runErr != nil {
+
+		//Check if the command timed out or exceeded memory limit
+		exitError := runErr.(*exec.ExitError)
+
+		e.logger.Infof("Command exited with status code %d [MsgID: %s]", exitError.ExitCode(), messageID)
+		if exitError.ExitCode() == ExitCodeTimeout {
+			e.logger.Errorf("Command timed out [MsgID: %s]", messageID)
+			timeOutContent := fmt.Sprintf("The command timed out after %d seconds\n", commandConfig.TimeLimit)
+			fmt.Fprint(stderr, timeOutContent)
+			return &ExecutionResult{
+				StatusCode: ErTimeout,
+				Message:    "The command timed out",
+			}
+		}
+
+		// Read stderr file
+		stderrContent, err := os.ReadFile(stderr.Name())
+		if err != nil {
+			e.logger.Errorf("Could not read stderr file. %s [MsgID: %s]", err.Error(), messageID)
+			return &ExecutionResult{
+				StatusCode: ErInternalError,
+				Message:    fmt.Sprintf("could not read stderr file. %s", err.Error()),
+			}
+		}
+		errorOutput := string(stderrContent)
+		e.logger.Errorf("Error executing command. %s [MsgID: %s]", errorOutput, messageID)
+
+		// Check if the command was blocked by chroot
+		if strings.Contains(errorOutput, "chroot") {
+			e.logger.Errorf("Command blocked due to chroot restrictions [MsgID: %s]", messageID)
+			return &ExecutionResult{
+				StatusCode: ErJailed,
+				Message:    fmt.Sprintf("command blocked due to chroot restrictions. %s", errorOutput),
+			}
 		}
 	}
 
 	var statusCode ExecutorStatusCode
-	switch cmd.ProcessState.ExitCode() {
+	switch restrictedCmd.ProcessState.ExitCode() {
 	case -1:
 		statusCode = ErSignalRecieved
 	case 0:
 		statusCode = ErSuccess
 	default:
-		logger.Errorf("Command exited with unknown status code. %d [MsgID: %s]", cmd.ProcessState.ExitCode(), messageID)
+		e.logger.Errorf("Command exited with unknown status code. %d [MsgID: %s]", restrictedCmd.ProcessState.ExitCode(), messageID)
 		return &ExecutionResult{
 			StatusCode: ErInternalError,
-			Message:    fmt.Sprintf("command exited with unknown status code. %d", cmd.ProcessState.ExitCode()),
+			Message:    fmt.Sprintf("command exited with unknown status code. %d", restrictedCmd.ProcessState.ExitCode()),
 		}
 	}
 
-	logger.Infof("Command executed successfully [MsgID: %s]", messageID)
+	e.logger.Infof("Command executed successfully [MsgID: %s]", messageID)
```

Comment:
The message of logger is not consistent. Because status code can be ErSignalRecieved, then this will not reflect what happened. Lets change this to "Finished executing command with status code: %d..."